lib/fog/orchestration/openstack/models/stack.rb

```diff
@@ -14,7 +14,11 @@ class Stack < Fog::OpenStack::Model
 
         def save(options = {})
           if persisted?
-            service.update_stack(self, default_options.merge(options)).body['stack']
+            stack_default_options = default_options
+            if [:template,:template_url].any? { |e| options.key?(e) }
```

Comment:
I think you only need to check for `:template_url` here. If `:template` is in the options then it will overwrite `:template` in `default_options` when the `merge(options)` is done in line 21, so there is no need to specifically remove it from `default_options`.